### PR TITLE
Add Widgets for Byte and Word Blocks

### PR DESCRIPTION
**Pull request by marsiliano**
Add Byte Widget
Add Word Widget
Update authors list

### CODE REVIEW

File: src/gui/ByteWidget.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class QSpinBox;
+
+class ByteWidget : public QWidget
+{
+    Q_OBJECT
+  public:
+    explicit ByteWidget(QString description = QString(),
+                        quint8 defaultValue = 0xFF, QWidget *parent = nullptr);
+
+  signals:
```
Comment (guerinoni) [inline src/gui/ByteWidget.hpp]: this?

File: src/gui/ByteWidget.hpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  public slots:
```
Comment (guerinoni) [inline src/gui/ByteWidget.hpp]: and this?

File: src/gui/ByteWidget.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    m_decValueLabel = new QLabel(this);
+    m_hexValueLabel = new QLabel(this);
+    int index       = 0;
+    layout->addWidget(m_valueSpinBox, 0, index++);
+    layout->addWidget(m_decValueLabel, 0, index++);
+    layout->addWidget(m_hexValueLabel, 0, index++);
+    check();
+}
+
+void ByteWidget::check()
```
Comment (guerinoni) [inline src/gui/ByteWidget.cpp]: What does it check???

File: src/gui/WordWidget.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    m_decValueLabel = new QLabel(this);
+    m_hexValueLabel = new QLabel(this);
+    int index       = 0;
+    layout->addWidget(m_valueSpinBox, 0, index++);
+    layout->addWidget(m_decValueLabel, 0, index++);
+    layout->addWidget(m_hexValueLabel, 0, index++);
+    check();
+}
+
+void WordWidget::check()
```
Comment (guerinoni) [inline src/gui/WordWidget.cpp]: Same

File: src/gui/WordWidget.hpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+class WordWidget : public QWidget
+{
+    Q_OBJECT
+  public:
+    explicit WordWidget(QString description  = QString(),
+                        quint16 defaultValue = 0xFFFF,
+                        QWidget *parent      = nullptr);
+
+  signals:
```
Comment (guerinoni) [inline src/gui/WordWidget.hpp]: same

File: src/gui/WordWidget.hpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  public slots:
```
Comment (guerinoni) [inline src/gui/WordWidget.hpp]: same

File: src/gui/gui.pro [hunk 7]
```diff
@@ -11,15 +11,19 @@ HEADERS += \
     $$PWD/ConfigViewFactory.hpp \
     $$PWD/DialogSerialSettings.hpp \
     $$PWD/MdiChild.hpp \
-    $$PWD/Led.hpp
+    $$PWD/Led.hpp \
+    ByteWidget.hpp \
```
Comment (guerinoni) [inline src/gui/gui.pro]: Missing $$PWD

File: src/gui/gui.pro [hunk 8]
```diff
@@ -11,15 +11,19 @@ HEADERS += \
...
 
 SOURCES += \
     $$PWD/MainWindow.cpp \
     $$PWD/DialogAbout.cpp \
     $$PWD/ConfigViewFactory.cpp \
     $$PWD/DialogSerialSettings.cpp \
     $$PWD/MdiChild.cpp \
-    $$PWD/Led.cpp
+    $$PWD/Led.cpp \
+    ByteWidget.cpp \
```
Comment (guerinoni) [inline src/gui/gui.pro]: same as above

**Review by guerinoni — CHANGES_REQUESTED** [review]

**Review by AndreaRicchi — COMMENTED** [review]
I quote the @guerinoni comments.

File: src/gui/ByteWidget.cpp (hunk 3 above)
Comment (marsiliano) [reply src/gui/ByteWidget.cpp]: I used the same methods used in LedWidget

**Pull request merged**